server/server_config.go

```diff
@@ -147,9 +147,17 @@ type ServerConfig struct {
 	AuthZStreamInterceptor grpc.StreamServerInterceptor
 
 	// ExternalAuthZChecker plugs into the external authorizer framework's authZ interceptor
-	ExternalAuthZChecker     ExternalAuthZChecker
+	ExternalAuthZChecker ExternalAuthZChecker
+
+	// InsecureNoAuthNAuthZReqs is data passed by the caller for the caller's
+	// interceptor containing information on what APIs to not check for
+	// authentication or authorization
 	InsecureNoAuthNAuthZReqs []interface{}
```

Comment:
Can we also point readers to this documentation which explains things in a bit more detail? Main point to drive home here is that these are while-listing the API request types that are insecure. This is a second level of confirmation that it is ok to bypass authN and/or authZ check for the API request. The first confirmation is from the return value of `GetAuthZRequest()` call.
```
// ExternalAuthZRequestGetter must be implemented by all gRPC services that use the external authorizer.
type ExternalAuthZRequestGetter interface {
	// GetAuthZRequest is invoked by the authZ interceptor before performing an authZ check.
	// Returns an auth request that will be passed to ExternalAuthZChecker function to authorize
	// the specified input API request. Optionally, returns handler-data to be stashed in the context
	// for the later retrieval by the handler. The first return param can have following special values:
	// - Return InsecureNoAuthNAuthZ to skip both authN and authZ completely.
	// - Return InsecureNoAuthZ to perform just an authN check for a specific request and skip authZ check.
	// Such insecure requests must also be whilelisted in insecureNoAuthNAuthZReqs or insecureNoAuthZReqs params.
	// ExternalAuthZChecker function is not invoked for the insecure requests.
	GetAuthZRequest(ctx context.Context, fullPath string, request interface{}) (ExternalAuthZRequest, HandlerData, error)
}
```

Reply:
Suggestion:

```
// InsecureNoAuthNAuthZReqs is a list of API request types for which AuthN or AuthZ checks are skipped.
// When ExternalAuthZRequestGetter.GetAuthZRequest() returns InsecureNoAuthNAuthZ, the framework
// ensures that the request type is also present in InsecureNoAuthNAuthZReqs  list. This adds a second level
// of confirmation that it is ok to skip the auth checks for this request. Refer to the documentation
// of ExternalAuthZRequestGetter  interface for more details.
```